mailbox/plugin/quota-mailing-cassandra/src/test/java/org/apache/james/mailbox/quota/cassandra/dto/DTOTest.java

```diff
@@ -0,0 +1,156 @@
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.mailbox.quota.cassandra.dto;
+
+import static net.javacrumbs.jsonunit.fluent.JsonFluentAssert.assertThatJson;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.apache.james.core.User;
+import org.apache.james.eventsourcing.EventId;
+import org.apache.james.eventsourcing.cassandra.JsonEventSerializer;
+import org.apache.james.mailbox.model.Quota;
+import org.apache.james.mailbox.quota.QuotaCount;
+import org.apache.james.mailbox.quota.QuotaSize;
+import org.apache.james.mailbox.quota.mailing.aggregates.UserQuotaThresholds;
+import org.apache.james.mailbox.quota.mailing.events.QuotaThresholdChangedEvent;
+import org.apache.james.mailbox.quota.model.HistoryEvolution;
+import org.junit.jupiter.api.Test;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.datatype.jdk8.Jdk8Module;
+
+class DTOTest {
+
+    public static final Quota<QuotaSize> SIZE_QUOTA = Quota.<QuotaSize>builder().used(QuotaSize.size(23)).computedLimit(QuotaSize.size(33)).build();
+    public static final Quota<QuotaCount> COUNT_QUOTA = Quota.<QuotaCount>builder().used(QuotaCount.count(12)).computedLimit(QuotaCount.count(45)).build();
+    public static final QuotaThresholdChangedEvent EVENT = new QuotaThresholdChangedEvent(
+        EventId.first(),
+        HistoryEvolution.noChanges(),
+        HistoryEvolution.noChanges(),
+        SIZE_QUOTA,
+        COUNT_QUOTA,
+        UserQuotaThresholds.Id.from(User.fromUsername("foo@bar.com")));
+
+    public static final String EVENT_JSON = "{" +
+        " \"type\": \"quota-threshold-change\"," +
+        " \"eventId\": 0," +
+        " \"user\": \"foo@bar.com\"," +
+        " \"sizeQuota\": {" +
+        "   \"used\": 23," +
+        "   \"limit\": 33" +
+        " }," +
+        " \"countQuota\": {" +
+        "   \"used\": 12," +
+        "   \"limit\": 45" +
+        " }," +
+        " \"sizeEvolution\": {" +
+        "   \"change\": \"NoChange\"," +
+        "   \"threshold\": null," +
+        "   \"instant\": null," +
+        "   \"recentness\": null" +
+        " }," +
+        " \"countEvolution\": {" +
+        "   \"change\": \"NoChange\"," +
+        "   \"threshold\": null," +
+        "   \"instant\": null," +
+        "   \"recentness\": null" +
+        " }" +
+        "}";
+
+    public static final String COUNT_QUOTA_JSON = "{" +
+        "   \"used\": 12," +
+        "   \"limit\": 45" +
+        " }";
+
+    public static final String NO_CHANGES_JSON = "{" +
+        "  \"recentness\":null," +
+        "  \"threshold\":null," +
+        "  \"instant\":null," +
+        "  \"change\":\"NoChange\"" +
+        "}";
+
+    @Test
+    void shouldSerializeQuotaDTO() throws Exception {
+        ObjectMapper objectMapper = new ObjectMapper();
```

Comment:
why do we keep repeating that 2 lines ? aren't we supposed to use the same as in our production code (with same config) ?

Reply:
I had many problems while writing the DTOs.

These tests help to know if the problem came from the DTOs or from the JsonDeserializer code.

But we can not directly use the JsonDeserializer as these are not events. Hence this code.

I'd like to keep these tests as they GREATLY helped me: they have value.

Reply:
I generally don't agree on such arguments : what have value during dev is not to be kept everytime.
We should keep as little tests as possible that add value to future devs.
As the solution is already in the production code, having some tests explaining how to use jackson API is a bit duplicate.
Finally, I'm not against keeping it, but just be careful about what we expect from unit test.